moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -212,24 +211,82 @@ const findDifferenceByFeatureProperty = (featureProperty, arrayOne, arrayTwo) =>
  * @property {function} setIsDrawing - Toggle draw tools
  * @property {function} renderMapDrawTools - Function that returns JSX for the draw tools in the map
  * @property {function} saveDrawnPoints - Function that saves features drawn in the UI
+ * @property {function} saveActionDispatch - Function that helps us send signals to other components
  */
 export function useMapDrawTools(
   featureCollection,
   setFeatureCollection,
   projectId,
   refetchProjectDetails,
-  currentZoom
+  currentZoom,
+  saveActionDispatch
 ) {
-  const isNewProject = projectId === null;
-
   const mapEditorRef = useRef();
   const [isDrawing, setIsDrawing] = useState(false);
   const [modeId, setModeId] = useState(null);
   const [modeHandler, setModeHandler] = useState(null);
-  const [selectedFeatureIndex, setSelectedFeatureIndex] = useState(null);
-  const [selectedEditHandleIndexes, setSelectedEditHandleIndexes] = useState(
-    []
-  );
+
+  /**
+   * Returns true if lineStringA has the same coordinates as lineStringB
+   * @param {Object} lineStringA - A line string feature object to compare
+   * @param {Object} lineStringB - Another line string feature object to compare against
+   * @return {boolean}
+   */
+  const lineStringEqual = (lineStringA, lineStringB) => {
+    const lineStringCoordinatesA = lineStringA?.geometry?.coordinates ?? [];
+    const lineStringCoordinatesB = lineStringB?.geometry?.coordinates ?? [];
+
+    return (
```

Comment:
This is probably a non-thing; I have a feeling that this isn't probably an issue as you would have bumped up against it in your development.

I've had times in the past when I was trying to compare geometries outside of a GIS library by comparing the floating point numbers that represent the X, Y coordinates of a vertex or point, and they would come back as not equal when they were effectively the same at a human scale. For example, take two latitude values, 30.26722873 == 30.26722874 may be true or false depending on how the language or library you're using handles floats, even though the actual distance on the ground of the is ~1 millimeter. 

I think John's suggestion about turfjs would totally make this a non-issue, since doing geometry comparisons is a core ability when working with spatial data. Just thinking out-loud - maybe they define a numeric type with a fixed precision and anything beyond whatever decimal they decide is the last significant digit is truncated and ignored.

Reply:
You totally right, this type of comparisons are way over my head I gotta leave it to the pros. I'd leave this up for debate, I would need to get familiar with leaf.js and then incorporate it and test it.